### DISCUSSION TITLE
gprc api

### DISCUSSION THREAD

**Discussion by sonic182** [ggml-org/llama.cpp#2101] [Ideas]
There is any repo like "llama-server" or similar? so we can have a llama.cpp instance running a model and a GRPC api for sending prompts and receiving completions?

**Comment by ianscrivener** [comment]
Llama.cpp has a simple [HTTP API server](https://github.com/ggerganov/llama.cpp/tree/master/examples/server)

There are a few other Bindings mentioned on the [README.md homepage of this repo](https://github.com/ggerganov/llama.cpp).

I use the **Python API by Abetlen** et all [abetlen/llama-cpp-python](https://github.com/abetlen/llama-cpp-python).. which very conveniently **mimics the OpenAI API**... so you can use any of the OpenAI ChatGPT clients >> llama-cpp-python >> llama.cpp.

Sorry, I don't of a GRPC API for llama.cpp
